cmd/frontend/graphqlbackend/saved_searches.go

```
@@ -111,20 +112,78 @@ func (r *schemaResolver) toSavedSearchResolver(entry types.SavedSearch) *savedSe
 	return &savedSearchResolver{db: r.db, s: entry}
 }
 
-func (r *schemaResolver) SavedSearches(ctx context.Context) ([]*savedSearchResolver, error) {
-	a := actor.FromContext(ctx)
-	if !a.IsAuthenticated() {
-		return nil, errors.New("no currently authenticated user")
+type savedSearchesArgs struct {
+	graphqlutil.ConnectionResolverArgs
+	Namespace graphql.ID
+}
+
+func (r *schemaResolver) SavedSearches(ctx context.Context, args savedSearchesArgs) (*graphqlutil.ConnectionResolver[savedSearchResolver], error) {
+	var userID, orgID int32
+	if err := UnmarshalNamespaceID(args.Namespace, &userID, &orgID); err != nil {
+		return nil, err
+	}
+
+	if userID != 0 {
+		if err := auth.CheckSiteAdminOrSameUser(ctx, r.db, userID); err != nil {
+			return nil, err
+		}
+	} else if orgID != 0 {
+		if err := auth.CheckOrgAccessOrSiteAdmin(ctx, r.db, orgID); err != nil {
+			return nil, err
+		}
+	}
```

What if both are `0`? I think we should return an error in that case.

---

fixed